polkadot/runtime/parachains/src/inclusion/mod.rs

```diff
@@ -1124,6 +1119,41 @@ impl<T: Config> Pallet<T> {
 	) -> Option<CandidatePendingAvailability<T::Hash, BlockNumberFor<T>>> {
 		<PendingAvailability<T>>::get(&para)
 	}
+
+	/// Attempts to schedule a code upgrade.
+	///
+	/// Returns the consumed weight.
+	fn try_schedule_code_upgrade(
+		config: &HostConfiguration<BlockNumberFor<T>>,
+		para_id: ParaId,
+		new_code: primitives::ValidationCode,
+	) -> Weight {
+		// Block number of candidate's inclusion.
+		let now = <frame_system::Pallet<T>>::block_number();
+
+		match <T as paras::Config>::PreCodeUpgrade::pre_code_upgrade(
+			para_id,
+			new_code.clone(),
+			UpgradeRequirements::EnforceRequirements,
+		) {
+			Ok(consumed_weight) =>
+				consumed_weight.saturating_add(<paras::Pallet<T>>::schedule_code_upgrade(
+					para_id,
+					new_code,
+					now,
+					&config,
+					SetGoAhead::Yes,
+				)),
+			Err(consumed_weight) => {
```

Comment:
We need to call the following code in `paras`:
```
	UpgradeGoAheadSignal::<T>::insert(&para_id, UpgradeGoAhead::Abort);
```

There should also be some test for this.

Reply:
Thanks for catching this!